src/link/MachO/Dylib.zig

```diff
@@ -74,6 +75,8 @@ pub fn closeFile(self: Dylib) void {
 pub fn parse(self: *Dylib) !void {
     log.debug("parsing shared library '{s}'", .{self.name.?});
 
+    var seek_pos = try self.file.?.getPos();
```

Comment:
Given `zld` assumes all the input mach-o files are 64bit ones I'd place the fat-parsing code here (or, better, in a subroutine that's called here).
The base offset can be computed once, according to the selected architecture, and then stored in `Dylib` to avoid passing `seek_pos` everywhere.

Reply:
It might be worth looking at how this is done in `Archive.zig` which after all is a collection of objects, so naturally we need to seek to find the right object.

Reply:
Oh and we should not have to parse *all* dylibs within a fat dylib, only the one matching the requested architecture.